src/io/iohk/scalanet/test/peergroup/TLSPeerGroupSpec.scala

```diff
@@ -0,0 +1,76 @@
+package io.iohk.scalanet.peergroup
+
+import io.iohk.decco.auto._
+import io.iohk.scalanet.NetUtils._
+import io.iohk.scalanet.TaskValues._
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.Matchers._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.{BeforeAndAfterAll, FlatSpec}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Random
+
+class TLSPeerGroupSpec extends FlatSpec with BeforeAndAfterAll {
+
+  implicit val patienceConfig: ScalaFutures.PatienceConfig = PatienceConfig(5 seconds)
+
+  behavior of "TCPPeerGroup"
+
+  it should "send and receive a message when client auth is disabled / false" in
+    withTwoRandomTLSPeerGroups[String](false) { (alice, bob) =>
```

Comment:
please can you test using both an authenticated and unauthenticated configuration (there is code you can borrow from DTLSPeerGroupSpec for that).